docs/vm-metrics.md

```diff
@@ -0,0 +1,254 @@
+  the development speed
+
+#### collectd
+The [collectd](http://collectd.org/) project fullfills all the KubeVirt monitoring needs, featuring
+a wealth of plugins including a Prometheus endpoint and one libvirt connector.
+Collectd is a well established monitoring solution and it is often mentioned in the examples of
+the Kubernetes documentation about setting a monitoring pipeline.
+
+**PROS**
+- Most of the functionalities ready
+- Established, mature solution
+- Small image size
+
+**CONS**
+- Reporting of ephemeral metrics could need work and acceptance upstream
+- Collectd uses one [implicit and hardcoded mapping](https://github.com/collectd/collectd/blob/master/src/write_prometheus.c#L405)
+  for prometheus labels which is a poor fit for the KubeVirt use case.
+- long release cycle
+
+## KubeVIRT API Adapters: VM metrics naming rules
+
+The task of the KubeVirt HPA API adapter is to expose the metrics collected by the cluster agent
+as custom metrics. The main challenge for this component is to provide a stable API and a naming
+convention for the VM metrics.
+
+Wherever possible **we should follow the existing naming conventions**. The following guidelines
+aims to make the metrics easier to use for the VM case, but are still under discussion.
+
+The additional requirement for VM metrics are
+- predictability of the names: names should use a common pattern as much as possible
+- avoid or at least minimize the clashes: wherever useful, it it should be possible to
+  disaggregate the metrics per-VM. For example, one could be interested in the metrics related to
+  the migration of one specific VM, to make sure the migration is converging and to take actions
+  if it is not.
+
+In order to minimize the clashes and make the VM metrics name predictable, we adopt the following rules
+to set the names:
+
+1. **TO BE CHECKED** If namespaces are available, they should be used: each VM has its own namespace - the VM name or UUID,
+   to be decided. Otherwise, a metric name should include the VM identifier (name or UUID) as prefix.
+
+2. If not obvious, the metric name should include the unit. Examples:
+
+- **BAD**: `VM0-migration-processedMemory`
+  What is it? bytes? KiBs? I need to look it up in the documentation.
+
+- **GOOD**: `VM0-migration-processedMemoryKiB`
+
+3. Keep the common prefixes among related metrics as long as possible. Examples
+
+- **BAD**: `VM0-migration-processedMemory`, `VM0-migration-remainingMemory`
+
+- **GOOD**: `VM0-migration-memoryProcessed`, `VM0-migration-memoryRemaining`
+
+Once name is exported, it is considered part of the API stability and treated accordingly.
+
+## Final diagram
+
+The following diagram illustrates how the KyubeVirt monitoring pipeline will look like
+with all the components described in this document deployed
+
+![VM metrics monitoring pipeline example](vm-metrics-example.png "VM Metrics monitoring pipeline example")
```

Comment:
Could you add existing component names (if they exist) to the boxes? E.g. "Monitoring CLuster Agent" probably relates to Heapster or Prometheus?